### PR TITLE
Fix java float/double parsing tests [skip ci]

### PR DESCRIPTION
**Pull request by revans2**
This is a part of #7410 and depends on it.  The java tests when they were first written, for what ever reason, did not use the correct string representation for several values including double max.  They ended up testing that the code was broken, which is not the right thing to do.  This updates those to have the correct values, but also has a comment about why the ideal value to test for overflow into `Inf` and `-Inf` is not ideal, but is a work around for small issues with the parsing code still.

### CODE REVIEW

**Review by jlowe — APPROVED** [review]

**Comment by revans2** [comment]
@gpucibot merge

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 1]
```diff
@@ -3229,30 +3229,45 @@ void testIsFloat() {
...
+  @Test
+  void testIsDouble() {
+    String[] doubleStrings = {"A", "nan", "Inf", "-Inf", "Infinity", "infinity", "-0.0", "0.0",
+        "1.7976931348623157E308",
+        // Current CUDF Code does not detect overflow for this. "1.7976931348623158E308",
+        // So we make it a little larger for this test
+        "1.7976931348623159E308",
+        "-1.7976931348623157E308",
+        // Current CUDF Code does not detect overflow for this. "-1.7976931348623158E308",
+        // So we make it a little larger for this test
```
Comment (hyperbolic2346) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:3254]: smaller?
